gpm_api/tests/test_io/test_data_integrity.py

```diff
@@ -20,7 +21,28 @@ def test_get_corrupted_filepaths(
         ), "Corrupted paths array should be the same length as input paths"
         assert abs_paths == res, "Corrupted paths array should be the same as input paths"
 
-    # TODO: Test an actual HDF5 for OSError and empty list (success)
+
+def test_get_corrupted_filepaths_real_files(
+    tmpdir: str,
+) -> None:
+    filepath = os.path.join(tmpdir, "test.h5")
+
+    # Create hdf5 file
+    array = xr.DataArray([])
+    array.to_netcdf(filepath)
+
+    # Test that no paths are "corrupted"
+    res = di.get_corrupted_filepaths([filepath])
+    assert len(res) == 0, "Corrupted paths array should be empty"
+
+    # Corrupt file by truncating it
+    with open(filepath, "r+") as f:
+        file_size = os.path.getsize(filepath)
+        f.truncate(round(file_size / 2))
```

Comment:
Good to know ...
Do we want to still add to the repo a real corrupted HDF5 file generated by interrupting the download?

Reply:
I don't think that it's necessary, the behavior would be the same (`h5py.File(filepath, "r")` failing)

Reply:
Ok